azure-iot-device/azure/iot/device/iothub/client_event.py

```diff
@@ -0,0 +1,15 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+NEW_SASTOKEN_REQUIRED = "NEW_SASTOKEN_REQUIRED"
+CONNECTION_STATE_CHANGE = "CONNECTION_STATE_CHANGE"
+BACKGROUND_EXCEPTION = "BACKGROUND_EXCEPTION"
+
+
+class ClientEvent(object):
```

Comment:
>ClientEvent [](http://example.com/codeflow?start=6&length=11)

doc please, especially if users receive one of these. (I'm not through the code far enough to know how these flow yet)

Reply:
They don't, but that's a good idea to document.

These are essentially just glorified tuples matching an event name with any associated values that may exist for the event and need to be given to the user.

Since they are created and used in different modules, it felt weird to just use tuples with semantics that were a mere convention. I figured a little structure would be a good idea for readability and maintenance going forth.